lib/constructs/iterable.js

```diff
@@ -34,40 +34,20 @@ class Iterable {
   }
 
   generate() {
-    const isPairIterator = utils.isPairIterable(this.idl);
     let str = "";
 
-    if (isPairIterator) {
+    if (this.obj.hasPairIterator) {
       str += `
-        ${this.generateFunction(Symbol.iterator, "key+value", "Symbol.iterator", "entries")}
         ${this.obj.name}.prototype.entries = ${this.obj.name}.prototype[Symbol.iterator];
         ${this.generateFunction("keys", "key")}
         ${this.generateFunction("values", "value")}
-        ${this.obj.name}.prototype.forEach = function forEach(callback) {
-          if (!this || !module.exports.is(this)) {
-            throw new TypeError("Illegal invocation");
-          }
-          if (arguments.length < 1) {
-            throw new TypeError("Failed to execute 'forEach' on '${this.obj.name}': 1 argument required, " +
-                                "but only 0 present.");
-          }
-          if (typeof callback !== "function") {
-            throw new TypeError("Failed to execute 'forEach' on '${this.obj.name}': The callback provided " +
-                                "as parameter 1 is not a function.");
-          }
-          const thisArg = arguments[1];
-          let pairs = Array.from(this[impl]);
-          let i = 0;
-          while (i < pairs.length) {
-            const [key, value] = pairs[i].map(utils.tryWrapperForImpl);
-            callback.call(thisArg, value, key, this);
-            pairs = Array.from(this[impl]);
-            i++;
-          }
-        };
       `;
     } else {
-      // value iterator; WIP
+      str += `
+        ${this.obj.name}.prototype.entries = Array.prototype.entries;
+        ${this.obj.name}.prototype.keys = Array.prototype.keys;
+        ${this.obj.name}.prototype.values = Array.prototype.values;
+      `;
```

Comment:
`entries`, `keys`, and `values` will only be present on an interface if the interface has an iterable declaration ([source]), while `Symbol.iterator` and `forEach` may be present in other situations [as well](https://heycam.github.io/webidl/#es-iterators). I thought it was more appropriate to put the things directly related to iterable declarations in `iterable.js`.